Sources/VLCBoxTableViewController.m

```diff
@@ -231,20 +232,32 @@ - (void)triggerDownloadForCell:(VLCCloudStorageTableViewCell *)cell
 
     if (_selectedFile.size.longLongValue < [[UIDevice currentDevice] VLCFreeDiskSpace].longLongValue) {
         /* selected item is a proper file, ask the user if s/he wants to download it */
-        VLCAlertView *alert = [[VLCAlertView alloc] initWithTitle:NSLocalizedString(@"DROPBOX_DOWNLOAD", nil) message:[NSString stringWithFormat:NSLocalizedString(@"DROPBOX_DL_LONG", nil), _selectedFile.name, [[UIDevice currentDevice] model]] delegate:self cancelButtonTitle:NSLocalizedString(@"BUTTON_CANCEL", nil) otherButtonTitles:NSLocalizedString(@"BUTTON_DOWNLOAD", nil), nil];
-        [alert show];
+        NSMutableArray<ButtonAction *> *buttonsAction = [[NSMutableArray alloc] init];
+        ButtonAction *cancelAction = [[ButtonAction alloc] initWithButtonTitle: NSLocalizedString(@"BUTTON_CANCEL", nil)
+                                                              buttonAction: ^(UIAlertAction* action){}];
+        ButtonAction *downloadAction = [[ButtonAction alloc] initWithButtonTitle:NSLocalizedString(@"BUTTON_DOWNLOAD", nil)
+                                                                    buttonAction:^(UIAlertAction* action){
+                                                                        [_boxController downloadFileToDocumentFolder:_selectedFile];
+                                                                        _selectedFile = nil;
+                                                                    }];
+        [buttonsAction addObject: cancelAction];
+        [buttonsAction addObject: downloadAction];
```

Comment:
Consider using a static array with the two items instead of an expensive mutable array where you only add items once during the life time of the object..